### DISCUSSION TITLE
Get minified version

### DISCUSSION THREAD

**Discussion by Chriton** [wevm/viem#384] [Question]
Hi, 

Is there a way to get a minified file that includes the entire dependencies so that I get use it like from a cdn in a <script> tag?
So far I have been trying to do this with adding this script  in package.json:

```pnpm run build && browserify ./dist/cjs/index.js | terser -c -m > bundle.min.js```

In theory this will generate a bundle.min.js file with everything included, but this doesn't seem to have all the dependencies included. 

Thanks!
<img width="1874" alt="Screenshot 2023-04-17 at 21 00 35" src="https://user-images.githubusercontent.com/12685981/232574636-fa0167a1-e5b3-47c5-affd-2275707e50b5.png">

<img width="1374" alt="Screenshot 2023-04-17 at 21 13 44" src="https://user-images.githubusercontent.com/12685981/232574750-85a0d08b-49e0-4749-bdff-110be0edf095.png">

**Comment by fubhy** [comment]
Hi. We currently do not publish a UMD bundle. Can you elaborate a bit on your use case?

I'm personally not particularly keen to add a UMD bundle to `viem` (ultimately that's the maintainers decision though). With the massive improvement in tooling & moderniation of the ecosystem that we've seen these past years, I can't see many (if any) good reasons to support UMD at this point.

**Reply by DerpHerpenstein** [reply]
We are writing a contract to turn EthFS into a eip4804 compliant CDN and We're exploring putting a viem on chain. The use case will be mostly artists and creators who want to put a basic UI for interacting with their contracts on chain, wherein the entirety of the data for their NFTs and websites is pulled directly from mainnet. We understand this is a niche use case.

**Comment by Chriton** [comment]
Any hints on how we can pack this in a UMD bundle in our own fork?

**Reply by fubhy** [reply]
You could probably use `tsup` https://tsup.egoist.dev/ with the `iife` format and a global name (`--format iife --global-name Viem`). I'd have to play around with the configuration myself first to give you a better answer. Another option would be to ues `rollup` directly.

**Comment by iammatthias** [comment]
Now that HTML editions are becoming more widespread, a minified file that includes the entire dependencies would be really nice. 

Currently the best way I've found to interact with onchain data from an html edition is to bundle web3.js. I'd much prefer to be using Viem.

**Reply by jxom** [reply]
here you go :) https://esm.sh/v126/viem/es2022/viem.bundle.mjs

**Reply by iammatthias** [reply]
You legend

**Reply by jxom** [reply]
https://twitter.com/jakemoxey/status/1674623675189039107?s=20
